artiq/frontend/artiq_coremgmt.py

```diff
@@ -10,6 +10,7 @@
 from artiq.coredevice.comm_kernel import CommKernel
 from artiq.coredevice.comm_mgmt import CommMgmt
 from artiq.coredevice.profiler import CallgrindWriter
+from artiq.frontend.artiq_flash import *
```

Comment:
Don't import from frontend, as the name suggests, this isn't for libraries.

Reply:
I think `artiq_coremgmt` needs to share some parts with `artiq_flash` so I have to import from frontend. `artiq_flash` also import from frontend.

Reply:
> I think artiq_coremgmt needs to share some parts with artiq_flash so I have to import from frontend. 

No, put the shared code somewhere else. See how code is shared between dashboard and browser for example.

> artiq_flash also import from frontend.

Yes, but it does not use it as a library and simply runs it equivalently to a command-line invocation, so it's somewhat acceptable.

Reply:
I see. So should I put these 3 functions in `tools.py`?

Reply:
Better not since they are very specific to the core device flash.

Reply:
Should I new a `flash.py` in coredevice folder?

Reply:
Normally this is for core device driver. I would suggest artiq/core_flash.py since there's already a remoting.py there with related functions. Maybe even merge remoting.py into core_flash.py.

Reply:
Merging `remoting.py` into another file may need to change a lot of import. I think it is better to add my functions into `remoting.py`

Reply:
Wrong reason, and it's not a lot of imports that need to be changed anyway.

Reply:
I searched and there is indeed only one import in artiq. So I'll new `artiq/core_flash.py` and merge `remoting.py` into it.